spyder/widgets/editor.py

```diff
@@ -0,0 +1,549 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# ----------------------------------------------------------------------------
+
+"""
+Simple code editor with syntax highlighting and line number area.
+
+Adapted from:
+https://doc.qt.io/qt-5/qtwidgets-widgets-codeeditor-example.html
+"""
+
+# Standard library imports
+import re
+
+# Third party imports
+from qtpy.QtCore import QRect, QSize, Qt, Signal
+from qtpy.QtGui import QColor, QPainter, QTextCursor, QTextFormat, QTextOption
+from qtpy.QtWidgets import QApplication, QPlainTextEdit, QTextEdit, QWidget
+
+# Local imports
+import spyder.utils.syntaxhighlighters as sh
+from spyder.widgets.mixins import BaseEditMixin
+
+
+# Constants
+LANGUAGE_EXTENSIONS = {

```

Comment:
You're repeating here almost the same values present in the `EDIT_FILETYPES` dictionary, placed in `spyder/config/utils.py`. So please do a bit of refactoring in that file, to be able to use the same file extensions here.

Reply:
>So please do a bit of refactoring in that file, to be able to use the same file extensions here.

Well, it is not really the same and it would require doing some changes that really feel out of scope since currently the SimpleCodeEditor just depends on syntax highlighting and base. Now it will depend on some values on a utils file in a config module. 

This can be addressed in the Editor migration since all these values should probably live there anyway. I think these "config.utils" should be elsewhere so I prefer to address the changes in a holistic fashion he the work arrises.